File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int n) {
    long long answer = 0;
    long long dp[1234567] = {0, 1, 2, };
    
    for(int i = 3; i <= n; i++) {
        dp[i] = dp[i - 2] + dp[i - 1] % 1234567;
    }
    answer = dp[n] % 1234567;
    return answer;
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    answer = 0
    dp = [0] * (n+1)
    dp[0], dp[1] = 1, 2
    
    for i in range(2, n):
        dp[i] = dp[i-2] + dp[i-1]
        
    answer = dp[n-1] % 1234567
    
    return answer

